packages/react-native-reanimated/src/threads.ts

```diff
@@ -255,3 +258,11 @@ export function runOnJS<Args extends unknown[], ReturnValue>(
     );
   };
 }
+
+// Override the logFunction implementation with the one that adds logs
+// with better stack traces to the LogBox
+loggerImpl.logFunction = (data: LogData) => {
+  'worklet';
+  runOnJS(logToLogBoxAndConsole)(data);
+};
+shareableMappingCache.set(logger, makeShareableCloneRecursive(logger));
```

Comment:
Let's export a function from `logger` called something like `replaceLoggerImplementation` since this might be a bit confusing.